File: backend/project-service/src/main/java/org/driventask/project/Payload/Response/ProjectResponse.java
package org.driventask.project.Payload.Response;

import java.time.LocalDateTime;
import java.util.Set;

public record ProjectResponse(
    String id,

    String name,
    
    String description,
    
    LocalDateTime startDate,
    
    LocalDateTime endDate,
    
    LocalDateTime createdAt,
    
    LocalDateTime updatedAt,
    
    Set<String> users_id,
    
    Set<String> task_id,

    Set<String> file_id
) {
}

File: frontend/src/app/models/project-response.ts
export interface ProjectResponse {
    id: string;
    name: string;
    description: string;
    startDate: Date;
    endDate: Date;
    status: string;
    createdAt: Date;
    updatedAt: Date;
    users_id: string[];
    task_id: string[];
    file_id: string[];
}